### DISCUSSION TITLE
How should one get an op's output and concatenate it with another op's output?

### DISCUSSION THREAD

**Discussion by ivanov-slk** [dagster-io/dagster#6585] [Q&A] [answered]
I'm using `dagster==0.13.19` on Linux and Python 3.9.

How should I take an `op`'s output and concatenate it with another `op`'s output? Let's suppose that both `op`s output `list`s.

Example code:
```
### zzz.py
from dagster import op, job

@op
def op_a(context):
    return [1,2,3]

@op
def op_b(context):
    return [4,5,6]

@job
def job_a():
    print(type(op_a))
    a = op_a()
    print(type(a))
    b = op_b()
    print(a+b)
```

When I execute this with `python zzz.py` I get the following error:
```
<class 'dagster.core.definitions.op_definition.OpDefinition'>
<class 'dagster.core.definitions.composition.InvokedSolidOutputHandle'>
Traceback (most recent call last):
  File "/home/myself/zzz.py", line 12, in <module>
    def job_a():
  File "/home/myself/.venv/lib/python3.9/site-packages/dagster/core/definitions/decorators/job.py", line 186, in job
    return _Job()(name)
  File "/home/myself/.venv/lib/python3.9/site-packages/dagster/core/definitions/decorators/job.py", line 63, in __call__
    ) = do_composition(
  File "/home/myself/.venv/lib/python3.9/site-packages/dagster/core/definitions/composition.py", line 1008, in do_composition
    output = fn(**kwargs)
  File "/home/myself/zzz.py", line 17, in job_a
    print(a+b)
TypeError: unsupported operand type(s) for +: 'InvokedSolidOutputHandle' and 'InvokedSolidOutputHandle'
```

I can see that `op`s return `InvokedSolidOutputHandle`. Is this okay? I thought that `solid`s were deprecated some time ago and one should be able to get the `op`'s output directly and use it in, e.g., addition etc.

In case this is desired, how should I concatenate these two lists?

Thank you!

**Comment by alangenfeld** [comment, marked as answer]
The `@job` decorated function runs only at initialization time to build the dependency structure. If you want to do any computation on data, it has to happen in an `@op`. 

https://docs.dagster.io/concepts/ops-jobs-graphs/jobs-graphs#fixed-fan-in may be relevant if your end goal is beyond just concatenating the outputs.

**Reply by ivanov-slk** [reply]
A classic example of [the XY problem](https://xyproblem.info/) on my side :D
I was indeed trying to put these two ops' outputs into a third op.

Anyway, thanks for guessing my real problem and the documentation link :)
